cni/network/multitenancy_test.go

```diff
@@ -61,12 +77,23 @@ func (c *MockCNSClient) ReleaseIPAddress(_ context.Context, ipconfig cns.IPConfi
 }
 
 func (c *MockCNSClient) GetNetworkContainer(ctx context.Context, orchestratorContext []byte) (*cns.GetNetworkContainerResponse, error) {
-	c.require.Exactly(c.getNetworkContainerConfiguration.orchestratorContext, orchestratorContext)
+	if !reflect.DeepEqual(c.getNetworkContainerConfiguration.orchestratorContext, orchestratorContext) {
+		return nil, errNoOrchestratorContextFound
+	}
 	return c.getNetworkContainerConfiguration.returnResponse, c.getNetworkContainerConfiguration.err
 }
 
 func (c *MockCNSClient) GetAllNetworkContainers(ctx context.Context, orchestratorContext []byte) ([]cns.GetNetworkContainerResponse, error) {
-	c.require.Exactly(c.getAllNetworkContainersConfiguration.orchestratorContext, orchestratorContext)
+	if _, isUnsupported := c.unsupportedAPIs[GetAllNetworkContainers]; isUnsupported {
+		e := &client.CNSClientError{}
+		e.Code = types.UnsupportedAPI
+		e.Err = errUnsupportedAPI
+		return nil, e
+	}
+
+	if !reflect.DeepEqual(c.getAllNetworkContainersConfiguration.orchestratorContext, orchestratorContext) {
```

Comment:
`reflect.DeepEqual` is rarely the correct func in a `*_test.go` file these days. The trouble is that it compares everything--unexported fields included. This is fine if you don't _have_ any unexported fields, but that may only be true today, at the time you're committing the code. It makes the tests more brittle because they make it more difficult for compared structs to have internals that are invisible to outside observers. Use `github.com/google/go-cmp/cmp` and the `cmp.Equal` func instead (which is a drop-in replacement). The upside is that you can provide diffs when they _aren't_ the same using `cmp.Diff`. That makes it much easier to hunt down broken things.

Reply:
fixed, use cmp.Equal